autotx/agents/TokenResearchAgent.py

```diff
@@ -0,0 +1,173 @@
+import json
+from textwrap import dedent
+from typing import Callable
+from crewai import Agent
+from autotx.AutoTx import AutoTx
+from autotx.auto_tx_agent import AutoTxAgent
+from autotx.utils.coingecko.api import CoingeckoApi
+from gnosis.eth import EthereumClient
+from crewai_tools import BaseTool
+
+coingecko = CoingeckoApi()
```

Comment:
I don't like how this is a global object instance. It doesn't look like the `CoingeckoApi` stores any state per-instance, so I think we can just make this a single function `coingecko_request` or something like that.

Reply:
> It doesn't look like the CoingeckoApi stores any state per-instance

that's why IMO it's not a bad practice to put it as a global variable - but sure i can make it a single function which returns this instance :smile: